From 7972f1ac1b85a5174c9f293ad98abc87edfe42e6 Mon Sep 17 00:00:00 2001
From: Maarten Coene <maartenc@apache.org>
Date: Sun, 23 Nov 2008 22:34:35 +0000
Subject: [PATCH] Fixed a problem with the regexp used for listing the possible
 values of a token, this is causing problems in combination with for instance
 the ivy:buildnumber task (IVY-970)

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@720059 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/ivy/plugins/resolver/util/ResolverHelper.java   | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/java/org/apache/ivy/plugins/resolver/util/ResolverHelper.java b/src/java/org/apache/ivy/plugins/resolver/util/ResolverHelper.java
index e0149482b..682da0c0b 100644
--- a/src/java/org/apache/ivy/plugins/resolver/util/ResolverHelper.java
+++ b/src/java/org/apache/ivy/plugins/resolver/util/ResolverHelper.java
@@ -72,10 +72,11 @@ public static String[] listTokenValues(Repository rep, String pattern, String to
                     } else {
                         namePattern = pattern.substring(slashIndex + 1);
                     }
-                    String acceptNamePattern = ".*"
+                    namePattern = namePattern.replaceAll("\\.", "\\\\.");
+                    String acceptNamePattern = ".*?"
                             + IvyPatternHelper.substituteToken(namePattern, token, "([^" + fileSep
-                                    + "]+)") + ".*";
-                    Pattern p = Pattern.compile(acceptNamePattern.toString());
+                                    + "]+)") + "($|" + fileSep + ".*)" ;
+                    Pattern p = Pattern.compile(acceptNamePattern);
                     for (Iterator iter = all.iterator(); iter.hasNext();) {
                         String path = (String) iter.next();
                         Matcher m = p.matcher(path);
